terraform/main.tf

```diff
@@ -0,0 +1,194 @@
+provider "aws" {
+  region = "us-east-1"
+}
+
+data "archive_file" "app_zip" {
+  type        = "zip"
+  source_dir  = "../app/dist"
+  output_path = "./app.zip"
+}
+
+resource "aws_apigatewayv2_api" "app" {
+  name          = "api"
+  protocol_type = "HTTP"
+}
+
+resource "aws_apigatewayv2_stage" "app" {
+  api_id = aws_apigatewayv2_api.app.id
+
+  name        = "api"
+  auto_deploy = true
+
+  access_log_settings {
+    destination_arn = aws_cloudwatch_log_group.api_gw.arn
+
+    format = jsonencode({
+      requestId               = "$context.requestId"
+      sourceIp                = "$context.identity.sourceIp"
+      requestTime             = "$context.requestTime"
+      protocol                = "$context.protocol"
+      httpMethod              = "$context.httpMethod"
+      resourcePath            = "$context.resourcePath"
+      routeKey                = "$context.routeKey"
+      status                  = "$context.status"
+      responseLength          = "$context.responseLength"
+      integrationErrorMessage = "$context.integrationErrorMessage"
+    }
+    )
+  }
+}
+
+resource "aws_apigatewayv2_integration" "app" {
+  api_id = aws_apigatewayv2_api.app.id
+
+  integration_uri    = aws_lambda_function.app.invoke_arn
+  integration_type   = "AWS_PROXY"
+  integration_method = "POST"
+}
+
+resource "aws_apigatewayv2_route" "app" {
+  api_id = aws_apigatewayv2_api.app.id
+
+  route_key = "ANY /{proxy+}"
+  target    = "integrations/${aws_apigatewayv2_integration.app.id}"
+}
+
+resource "aws_cloudwatch_log_group" "api_gw" {
+  name = "/aws/api_gw/${aws_apigatewayv2_api.app.name}"
+
+  retention_in_days = 30
+}
+
+resource "aws_lambda_permission" "api_gw" {
+  statement_id  = "AllowExecutionFromAPIGateway"
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.app.function_name
+  principal     = "apigateway.amazonaws.com"
+
+  source_arn = "${aws_apigatewayv2_api.app.execution_arn}/*/*"
+}
+
+resource "aws_security_group" "allow_db" {
+  name        = "allow_db"
+  description = "Allow DB"
+
+  ingress {
+    from_port        = 5430
+    to_port          = 5440
+    protocol         = "tcp"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+  egress {
+    from_port        = 0
+    to_port          = 0
+    protocol         = "-1"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+}
+
+resource "random_password" "password" {
+  length           = 20
+  special          = false
+  override_special = "_%@"
+}
+
+resource "aws_db_instance" "default" {
```

Comment:
Why PostgreSQL? :)

Could you elaborate on the reasoning behind picking this storage mechanism, and the pros and cons of the choice in a lambda-based application?

Reply:
As I know by requirement I can use any persistent DB. So Postgres is a good choice I think)
 
Regarding lambda, I know a lot of things.
1 First request is cold. so it can be slow. That's why you need to warm up it to decrease a cold start
2 You need to configure concurrency of lambda execution
3 It is expensive if you have high load - k8s better
4 You can easily integrate lambda to a lot of AWS services